src/main/zapHomeFiles/hud/serviceworker.js

```diff
@@ -136,10 +138,24 @@ const onMessage = event => {
 			break;
 
 		case "heartbeat":
-			webSocket.send('{ "component" : "hud", "type" : "view", "name" : "heartbeat" }');
+			apiCall("hud", "view", "heartbeat");
 			break;
 
+		case "zapApiCall":
+			if (event.ports.length > 0) {
+				apiCallWithResponse(message.component, message.type, message.name, message.params).then (response => {
```

Comment:
the style throughout the rest of the code is to have the `.then(response => {` start on the next line. helps to line up the promises. 

eventually all promises should be handled with:
```
let response = await apiCallWithResponse(...);
event.ports[0].postMessage(response);
```
which is obviously way nicer.

Reply:
will realign